prow/config/config.go

```diff
@@ -1257,6 +1257,8 @@ func validateAgent(v JobBase, podNamespace string) error {
 	case *v.Namespace != podNamespace && agent != b && agent != p:
 		// TODO(fejta): update plank to allow this (depends on client change)
 		return fmt.Errorf("namespace customization requires agent: %s or %s (found %q)", b, p, agent)
+	case v.RerunPermissions != nil && v.RerunPermissions.AllowAnyone && (v.RerunPermissions.Users != nil || v.RerunPermissions.Groups != nil):
```

Comment:
This doesn't have anything to do with the agent so please move this to a different validation function. (`validateJobBase`)

nit: Use `len(...) > 0` instead of `... != nil`.